Lib/test/test_pathlib.py

```diff
@@ -246,6 +136,46 @@ class P(_BasePurePathSubclass, self.cls):
         for parent in p.parents:
             self.assertTrue(parent.init_called)
 
+    def _check_drive_root_parts(self, arg, *expected):
+        def f(parts):
+            path = self.cls(*parts)
+            return path.drive, path.root, path.parts
+        sep = self.flavour.sep
+        altsep = self.flavour.altsep
+        actual = f([x.replace('/', sep) for x in arg])
+        self.assertEqual(actual, expected)
+        if altsep:
+            actual = f([x.replace('/', altsep) for x in arg])
+            self.assertEqual(actual, expected)
```

Comment:
Given that _check is called over 50 times, what do you think of pulling out the function to be created just once, and rename it at the same time.
```suggestion
    def _get_drive_root_parts(self, parts):
        path = self.cls(*parts)
        return path.drive, path.root, path.parts

    def _check_drive_root_parts(self, arg, *expected):
        sep = self.flavour.sep
        altsep = self.flavour.altsep
        actual = _get_drive_root_parts([x.replace('/', sep) for x in arg])
        self.assertEqual(actual, expected)
        if altsep:
            actual = _get_drive_root_parts([x.replace('/', altsep) for x in arg])
            self.assertEqual(actual, expected)
```
I personally would remove `altsept = ...` and change `if altsep:` to `if altsep := self.flavour.altset:`.  If you don't like `:=`, I would move the assignment to just before it is needed.  But if you prefer it where it is, leave it.